solutions/usaco-856.mdx

```diff
@@ -39,6 +39,54 @@ file_out.close()
 ```
 
 </PySection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class TheBucketList {
+	static final int maxTime = 1000;
+
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("blist");
+		int n = io.nextInt();
+
+		int[] startTimes = new int[n + 1];
+		int[] endTimes = new int[n + 1];
+		int[] bucketsNeeded = new int[n + 1];
+
+		for (int x = 1; x <= n; x++) {
+			startTimes[x] = io.nextInt();
+			endTimes[x] = io.nextInt();
+			bucketsNeeded[x] = io.nextInt();
+		}
+
+		// keep track of the maximum buckets in all of the times
+		int maxBuckets = 0;
+
+		// loop through each time and get the buckets needed
+		for (int time = 1; time <= maxTime; time++) {
+			int bucketsAtTime = 0;
+
+			// loop through and sum up the buckets for those with
+			// times in the range
+			for (int i = 1; i <= n; i++) {
+				if (startTimes[i] <= time && time <= endTimes[i]) {
+					bucketsAtTime += bucketsNeeded[i];
+				}
+			}
+			maxBuckets = Math.max(maxBuckets, bucketsAtTime);
+		}
+
+		io.println(maxBuckets);
+		io.close();
+	}
+}
+```
+	
```

Comment:
```suggestion
```

Reply:
no i think you need the space, in the previewer it is messed up without it

Reply:
i think the blank line is not technically necessary for the website, but is needed for syntax highlighting in some editors so i usually leave a blank line. it doesn't really matter though